From: Steve Dickson <SteveD@redhat.com>
Date: Mon, 1 Feb 2010 20:44:34 -0500
Subject: [nfs] Fix a reference leak in nfs_wb_cancel_page()
Message-id: <4B673D32.6000900@RedHat.com>
Patchwork-id: 23061
O-Subject: [RHEL6] [PATCH 1/8] NFS: Fix a reference leak in nfs_wb_cancel_page()
Bugzilla: 560784
RH-Acked-by: Peter Staubach <staubach@redhat.com>
RH-Acked-by: Jeff Layton <jlayton@redhat.com>

commit c9edda7140ec6a22accf7f2f86da362dfbfd41fc
Author: Trond Myklebust <Trond.Myklebust@netapp.com>
Date:   Tue Jan 26 15:41:34 2010 -0500

    NFS: Fix a reference leak in nfs_wb_cancel_page()

    Signed-off-by: Trond Myklebust <Trond.Myklebust@netapp.com>
    Cc: stable@kernel.org
    Reviewed-by: Chuck Lever <chuck.lever@oracle.com>

Signed-off-by: Aristeu Rozanski <arozansk@redhat.com>

diff --git a/fs/nfs/write.c b/fs/nfs/write.c
index 6d40a53..80722f4 100644
--- a/fs/nfs/write.c
+++ b/fs/nfs/write.c
@@ -1541,6 +1541,7 @@ int nfs_wb_page_cancel(struct inode *inode, struct page *page)
 			break;
 		}
 		ret = nfs_wait_on_request(req);
+		nfs_release_request(req);
 		if (ret < 0)
 			goto out;
 	}
